src/cmd/pkg/converter/terraform.go

```diff
@@ -0,0 +1,254 @@
+package converter
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/chronosphereio/chronoctl-core/src/cmd/pkg/sliceutil"
+	"github.com/chronosphereio/chronoctl-core/src/generated/swagger/configv1/models"
+	"github.com/chronosphereio/terraform-provider-chronosphere/chronosphere"
+	"github.com/chronosphereio/terraform-provider-chronosphere/chronosphere/intschema"
+	tfmodels "github.com/chronosphereio/terraform-provider-chronosphere/chronosphere/pkg/configv1/models"
+	"github.com/chronosphereio/terraform-provider-chronosphere/chronosphere/tfid"
+)
+
+// TerraformOutput are outputs of prometheus/alertmanager conversion that can be marshalled to HCL.
+type TerraformOutput struct {
+	NotificationPolicy *intschema.NotificationPolicy
+	Monitors           []*intschema.Monitor
+	Collections        []*intschema.Collection
+	RecordingRules     []*intschema.RecordingRule
+	PagerdutyNotifiers []*intschema.PagerdutyAlertNotifier
+	SlackNotifiers     []*intschema.SlackAlertNotifier
+	WebhookNotifiers   []*intschema.WebhookAlertNotifier
+	EmailNotifiers     []*intschema.EmailAlertNotifier
+	OpsGenieNotifiers  []*intschema.OpsgenieAlertNotifier
+	VictorOpsNotifiers []*intschema.VictoropsAlertNotifier
+}
+
+// ToTerraform converts the configv1 output of prometheus/alertmanager conversion into Terraform types that can be
+// marshalled to HCL. All resource references are resolved as <resource_type>.<resource_name>.<resource_attribute>
+// except for references to teams, which are resolved as <team_slug>. Teams are the exception because their Terraform
+// definitions are not generated via prom conversion and may live in separate modules.
+func ToTerraform(alerts *ConvertedAlerts) (*TerraformOutput, error) {
+	tfOut := &TerraformOutput{}
+
+	notifierTypes := make(map[string]string, len(alerts.Receivers))
+	for _, r := range alerts.Receivers {
+		notifier, err := tfNotifier(r.Configv1Notifier)
+		if err != nil {
+			return nil, fmt.Errorf("error converting notifier to Terraform: %v", err)
+		}
+		switch {
+		case notifier.Pagerduty != nil:
+			tf, err := chronosphere.PagerdutyAlertNotifierFromModel(&notifier)
+			if err != nil {
+				return nil, fmt.Errorf("error converting pagerduty notifier to Terraform: %v", err)
+			}
+			tf.HCLID = tfid.SafeID(notifier.Slug)
+			tfOut.PagerdutyNotifiers = append(tfOut.PagerdutyNotifiers, tf)
+			notifierTypes[notifier.Slug] = "chronosphere_pagerduty_alert_notifier"
+		case notifier.Email != nil:
+			tf, err := chronosphere.EmailAlertNotifierFromModel(&notifier)
+			if err != nil {
+				return nil, fmt.Errorf("error converting email notifier to Terraform: %v", err)
+			}
+			tf.HCLID = tfid.SafeID(notifier.Slug)
+			tfOut.EmailNotifiers = append(tfOut.EmailNotifiers, tf)
+			notifierTypes[notifier.Slug] = "chronosphere_email_alert_notifier"
```

Comment:
can we avoid these hardcoded values by using `.Ref()` and using `.Type()` (will probably need a typecast)
https://github.com/chronosphereio/terraform-provider-chronosphere/blob/main/chronosphere/intschema/classic_dashboard.go#L54